### PR TITLE
[skip-ci] RPM: create podman-machine subpackage

### PR DESCRIPTION
**Pull request by lsm5**
This subpackage will only fetch dependencies for podman-machine.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: rpm/podman.spec [hunk 1]
```diff
@@ -186,6 +186,16 @@ capabilities specified in user quadlets.
 It is a symlink to %{_bindir}/%{name} and execs into the `%{name}sh` container
 when `%{_bindir}/%{name}sh` is set as a login shell or set as os.Args[0].
 
+%define summary_machine Metapackage to install dependencies for %{name} machine
+
+%package machine
+Summary: %{summary_machine}
+Requires: %{name} = %{epoch}:%{version}-%{release}
+Requires: virtiofsd
```
Comment (Luap99) [inline rpm/podman.spec:194]: This should also add qemu and remove the Recommends: gvisor-tap-vsock and make it a requires here as this is also only used for mahcine

**Comment by cgwalters** [comment]
See https://github.com/containers/podman/pull/22920#discussion_r1629737267 for the sub-thread that introduced the requirement on `/usr/libexec/podman/virtiofsd`.

File: rpm/podman.spec [hunk 2]
```diff
@@ -186,6 +180,17 @@ capabilities specified in user quadlets.
...
+%package machine
+Summary: Metapackage for setting up %{name} machine
+Requires: %{name} = %{epoch}:%{version}-%{release}
+Requires: gvisor-tap-vsock
+Requires: qemu
+Requires: virtiofsd
+
+%description machine
+This subpackage installs the dependencies for %{name} machine and also creates
+a symlink to virtiofsd in %{_libexecdir}/%{name}
```
Comment (cgwalters) [inline rpm/podman.spec:192]: I don't think any user would care that it ships a symlink, that's an implementation detail.
How about:

> This subpackage installs the dependencies for "%{name} machine"; for more see https://docs.podman.io/en/latest/markdown/podman-machine.1.html

or something?

**Review by cgwalters — APPROVED** [review]
Looks sane to me!

File: rpm/podman.spec (hunk 2 above)
Comment (lsm5) [reply rpm/podman.spec:192]: i'll update, thanks

Comment (lsm5) [reply rpm/podman.spec]: done.

File: rpm/podman.spec (hunk 1 above)
Comment (lsm5) [reply rpm/podman.spec:188]: done

**Pull request merged**